test/Mono.Linker.Tests.Cases/DataFlow/NullableAnnotations.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using System.Diagnostics.CodeAnalysis;
+using Mono.Linker.Tests.Cases.Expectations.Helpers;
+using DAM = System.Diagnostics.CodeAnalysis.DynamicallyAccessedMembersAttribute;
+using DAMT = System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes;
+
+namespace Mono.Linker.Tests.Cases.DataFlow
+{
+	[SkipKeptItemsValidation]
+	[ExpectedNoWarnings]
+	class NullableAnnotations
+	{
+		struct TestStruct
+		{
+			public string FirstName { get; set; }
+			public string LastName { get; set; }
+		}
+
+		struct TestStructWithRucMethod
+		{
+			public string FirstName { get; set; }
+			public string LastName { get; set; }
+			[RequiresUnreferencedCode ("message")]
+			void MethodWithRuc () { }
+		}
+
+		class TestClass
+		{
+			public string FirstName { get; set; }
+			public string LastName { get; set; }
+		}
+
+		class NullableField
+		{
+			static TestClass? nullableTestClass;
+		}
+
+		public static void Main ()
+		{
+			NullableOfAnnotatedGenericParameter<TestStruct> ();
+			Type hasProperties = ReturnUnderlyingTypeThatRequiresProperties<Nullable<TestStruct>> (new ());
+			RequireProperties (hasProperties);
+
+			UnderlyingTypeOfUnannotatedGenericParameterRequiresProperties<Nullable<TestStruct>> ();
+
+			TestStruct? a = new TestStruct ();
+			RequiresPropertiesFromInstance (a);
+			TestClass? b = new TestClass ();
+			RequiresPropertiesFromInstance (b);
+			RequiresPropertiesFromInstance (new { Id = "asdf", Number = 18 });
+
+			UnderlyingTypeOfAnnotatedParameterRequiresAll (typeof (Nullable<TestStruct>));
+			RequireMethodWithRUC ();
+
+		}
+
+		[ExpectedWarning ("IL2026", "message")]
+		static void RequireMethodWithRUC ()
+		{
+			var T = typeof (Nullable<TestStructWithRucMethod>);
+			var uT = Nullable.GetUnderlyingType (T);
+			uT.RequiresAll ();
+		}
+
+		static void RequireProperties ([DAM (DAMT.PublicProperties)] Type t) { }
+
+		static void RequireAll ([DAM (DAMT.All)] Type t) { }
+
+		static void UnderlyingTypeOfAnnotatedParameterRequiresAll ([DAM (DAMT.All)] Type t)
+		{
+			RequireAll (Nullable.GetUnderlyingType (t));
+		}
+
+		static void NullableOfAnnotatedGenericParameter<[DAM (DAMT.PublicProperties)] T> () where T : struct
+		{
+			RequireProperties (Nullable.GetUnderlyingType (typeof (Nullable<T>)));
```

Comment:
I'm surprised this doesn't warn... - hmmm

Reply:
Thanks for pointing this out, it looks like in this case the Nullable<T> is still represented by SystemTypeValue, not NullableSystemTypeValue, so that needs to be fixed.

Reply:
Can you please also add a negative test - that if the inner T is not annotated this generates a warning?